src/skin/qml/qmllibraryproxy.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <QObject>
+#include <QString>
+
+class Library;
+class SidebarModel;
+
+QT_FORWARD_DECLARE_CLASS(QAbstractItemModel);
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+
+class QmlLibraryProxy : public QObject {
+    Q_OBJECT
+  public:
+    explicit QmlLibraryProxy(Library* pLibrary, QObject* parent = nullptr);
+
+    Q_INVOKABLE QAbstractItemModel* getSidebarModel();
+
+  private:
+    Library* m_pLibrary;
```

Comment:
How about using `QPointer` or `mixxx::SafeQPointer` to prevent dangling pointers?